### PR TITLE
Avoid generating the same ips for e2e

### PR DESCRIPTION
**Pull request by abhinavmpandey08**
*Issue #, if available:*

*Description of changes:*

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Comment by abhay-krishna** [comment]
[APPROVALNOTIFIER] This PR is **APPROVED**

This pull-request has been approved by: *<a href="https://github.com/aws/eks-anywhere/pull/654#" title="Author self-approved">abhinavmpandey08</a>*

The full list of commands accepted by this bot can be found [here](https://go.k8s.io/bot-commands?repo=aws%2Feks-anywhere).

The pull request process is described [here](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process)

<details >
Needs approval from an approver in each of these files:

- ~~[OWNERS](https://github.com/aws/eks-anywhere/blob/main/OWNERS)~~ [abhinavmpandey08]

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Review by jaxesn — APPROVED** [review]
Thanks!

**Comment by abhay-krishna** [comment]
[APPROVALNOTIFIER] This PR is **APPROVED**

This pull-request has been approved by: *<a href="https://github.com/aws/eks-anywhere/pull/654#" title="Author self-approved">abhinavmpandey08</a>*, *<a href="https://github.com/aws/eks-anywhere/pull/654#pullrequestreview-809016791" title="Approved">jaxesn</a>*

The full list of commands accepted by this bot can be found [here](https://go.k8s.io/bot-commands?repo=aws%2Feks-anywhere).

The pull request process is described [here](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process)

<details >
Needs approval from an approver in each of these files:

- ~~[OWNERS](https://github.com/aws/eks-anywhere/blob/main/OWNERS)~~ [abhinavmpandey08,jaxesn]

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

File: internal/test/e2e/uniqueip.go
Comment (g-gaston) [inline internal/test/e2e/uniqueip.go:9]: This is not thread safe :(
Is there a way we can do this before fanning out into multiple go routines? And just pass the ip to each "worker"

If not, we can get around it using a rw mutex
However, I would suggest not using a global variable. Just encapsulate it in a type and pass the same struct to the workers

**Review by g-gaston — COMMENTED** [review]
I like this

File: internal/test/e2e/uniqueip.go
Comment (abhinavmpandey08) [reply internal/test/e2e/uniqueip.go]: I have moved the logic for generating IPs before threads come into play

**Review by g-gaston — APPROVED** [review]
/lgtm

**Pull request merged**
